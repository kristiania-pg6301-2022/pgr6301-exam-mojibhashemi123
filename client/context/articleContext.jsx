import React from "react";
import { fetchJSON } from "../lib/fetchJSON";
import { postJSON } from "../lib/postJSON";

export const articleContext = React.createContext({
  async fetchArticle() {
    return await fetchJSON("/api/article");
  },

  async createArticle(article) {
    return await postJSON("/api/article", article);
  },
});
