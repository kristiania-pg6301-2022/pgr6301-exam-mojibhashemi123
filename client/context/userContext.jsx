import React from "react";
import { postJSON } from "../lib/postJSON";

export const UserContext = React.createContext({
  async createAccount(account) {
    return await postJSON("/api/register", account);
  },

  async loginSession(token) {
    return await postJSON("/api/register/login", token);
  },
});
