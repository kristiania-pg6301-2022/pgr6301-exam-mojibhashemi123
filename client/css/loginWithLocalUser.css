div .loginDiv{
    width: 180px;
    background-color: aqua;
    padding:8px;
    border: 2px;
    margin-bottom: 13px;
}