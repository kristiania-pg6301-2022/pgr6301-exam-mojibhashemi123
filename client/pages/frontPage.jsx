import React from "react";
import { Link } from "react-router-dom";

export function FrontPage() {
  return (
    <div id="frontpage">
      <h1>news article</h1>
      <ul>
        <li>
          <Link to={"/"}>Show news article</Link>
        </li>
        <li>
          <Link to={"/article/new"}>Add new article</Link>
        </li>
        <li>
          <Link to={"/article/update"}>Update your Article</Link>
        </li>
      </ul>
    </div>
  );
}
