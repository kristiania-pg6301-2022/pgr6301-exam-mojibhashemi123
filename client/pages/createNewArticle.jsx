import React, { useContext, useState } from "react";
import { useNavigate } from "react-router-dom";
import { FormInput } from "../lib/formInput";
import { articleContext } from "../context/articleContext";

export function CreateNewArticle({ user }) {
  const { createArticle } = useContext(articleContext);
  const [title, setTitle] = useState("");
  const [author, setAuthor] = useState("");
  const [topic, setTopic] = useState("");
  const [text, setText] = useState("");
  const [username, setUsername] = useState(user.microsoft?.name);
  const [error, setError] = useState();

  const navigate = useNavigate();

  async function handleSubmit(event) {
    event.preventDefault();
    const result = createArticle({
      title,
      author,
      topic,
      text,
      username,
    });

    try {
      if (title === "" || author === "" || topic === "" || text === "") {
        setError("Du må fylle");
        return;
      }
      await result;
      navigate("/");
    } catch (error) {
      console.error("ERROR: ", error);
      setError("Du er ikke pålogget");
    }
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <h1>Add new article</h1>
        <FormInput label={"Title:"} value={title} onChangeValue={setTitle} />
        <FormInput label={"Topics:"} value={topic} onChangeValue={setTopic} />
        <FormInput label={"Author:"} value={author} onChangeValue={setAuthor} />
        <strong>Text area:</strong>
        <textarea value={text} onChange={(e) => setText(e.target.value)} />
        {error && <div>{error}</div>}
        <div>
          <button>Submit</button>
        </div>
      </form>
    </div>
  );
}
