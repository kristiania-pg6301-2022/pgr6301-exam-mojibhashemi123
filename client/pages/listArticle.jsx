import { useContext, useEffect, useState } from "react";
import { articleContext } from "../context/articleContext";
import { useLoading } from "../useLoading";
import { Link, Route, Routes } from "react-router-dom";
let articles = [];
export function ListArticle() {
  const { fetchArticle } = useContext(articleContext);

  const { data, error, loading, reload } = useLoading(fetchArticle);

  if (loading) {
    return <div>Loading...</div>;
  }
  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  function handleClick() {
    document.getElementById("sidebar").classList.toggle("active");
  }

  function ListSideBar(event) {
    document.getElementById("sidebar").classList.toggle("dd");
    if (event) {
      console.log(event);

      articles = [];

      data.map((article, index) => {
        if (article.topic === event) {
          document.getElementById("sidebar").classList.toggle("dd");
          console.log(article.title + " fra sidebar");
          articles.push(article);
          //return ShowMatch(articles);
        }
      });
    }
  }
  let listTopic = Array.from(new Set(data.map((t) => t.topic)));

  return (
    <div id="#container2">
      <div id="sidebar" onClick={({ target }) => handleClick(target)}>
        <h1>Articles</h1>
        <div className={"toggle-btn"}>
          <span></span>
          <span></span>
          <span></span>
        </div>
        <ul>
          {listTopic.map((topic, index) => (
            <li onClick={({ target }) => ListSideBar(topic)}>
              <Link to={"/ShowMatch"}>{topic}</Link>
            </li>
          ))}
        </ul>
      </div>
      <h1></h1>
    </div>
  );
}

export function ShowMatch() {
  return (
    <div id="container2">
      {articles.map((articles, index) => (
        //for feil meldinger i browser bruk index
        <ArticleCard key={index} articles={articles} />
      ))}
    </div>
  );
}

function ArticleCard({ articles: { title, author, topic, text, date } }) {
  return (
    <>
      <h3>Title: {title}</h3>
      <h4>Author: {author}</h4>
      <div>Topic: {topic}</div>
      <p>Text: {text}</p>
      <h6>Date: {date}</h6>
      <p>
        -------------------------------------------------------------------------------------------------------------------------------------------------------------
      </p>
    </>
  );
}
