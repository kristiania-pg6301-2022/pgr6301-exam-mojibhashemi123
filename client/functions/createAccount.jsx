import { FormInput } from "../lib/formInput";
import { useContext, useState } from "react";
import { UserContext } from "../context/userContext";
import { useNavigate } from "react-router-dom";

export function CreateAccount() {
  const { createAccount } = useContext(UserContext);
  const [name, setName] = useState("");
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const navigate = useNavigate();

  function handleSubmit(event) {
    event.preventDefault();
    createAccount({ name, username, email, password });
    navigate("/login");
  }

  return (
    <form onSubmit={handleSubmit}>
      <h1>Add new movie</h1>

      <FormInput label={"First Name:"} value={name} onChangeValue={setName} />
      <FormInput
        label={"Username:"}
        value={username}
        onChangeValue={setUsername}
      />
      <FormInput label={"Email:"} value={email} onChangeValue={setEmail} />
      <FormInput
        label={"Password:"}
        value={password}
        onChangeValue={setPassword}
      />

      <div>
        <button>Submit</button>
      </div>
    </form>
  );
}
