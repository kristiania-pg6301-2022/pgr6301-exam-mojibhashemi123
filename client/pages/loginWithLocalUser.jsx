import React, { useContext, useState } from "react";
import { useNavigate } from "react-router-dom";
import { UserContext } from "../context/userContext";
import { FormInput } from "../lib/formInput";

import "../css/loginWithLocalUser.css";

export function LoginWithLocalUser() {
  const { loginSession } = useContext(UserContext);
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState();

  const navigate = useNavigate();

  async function handleSubmit(event) {
    event.preventDefault();
    const result = loginSession({ email, password });

    try {
      await result;
      navigate("/");
    } catch (error) {
      console.error("ERROR: ", error);
      setError("Du har tastet feil eller m√• du registerere deg");
    }
  }

  return (
    <div className={"loginDiv"}>
      <form onSubmit={handleSubmit}>
        <FormInput label={"Email:"} value={email} onChangeValue={setEmail} />
        <FormInput
          label={"Password:"}
          value={password}
          onChangeValue={setPassword}
        />
        {error && <div>{error}</div>}
        <div>
          <button>Log in</button>
        </div>
      </form>
    </div>
  );
}
