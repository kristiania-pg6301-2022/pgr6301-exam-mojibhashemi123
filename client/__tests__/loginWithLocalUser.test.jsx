import { LoginWithLocalUser } from "../pages/loginWithLocalUser";
import ReactDOM from "react-dom";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { UserContext } from "../context/userContext";
import { CreateAccount } from "../pages/createAccount";
import { Simulate } from "react-dom/test-utils";

jest.mock("../css/loginWithLocalUser.css", () => jest.fn());

describe("login component", () => {
  it("shows local login form", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <LoginWithLocalUser />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
    expect(
      Array.from(element.querySelectorAll("form label strong")).map(
        (e) => e.innerHTML
      )
    ).toEqual(["Email:", "Password:"]);
  });
});
