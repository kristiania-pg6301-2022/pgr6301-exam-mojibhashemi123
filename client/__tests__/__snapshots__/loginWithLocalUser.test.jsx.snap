// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component shows local login form 1`] = `"<div class=\\"loginDiv\\"><form><div class=\\"form-input\\"><label><strong>Email:</strong> <input value=\\"\\"></label></div><div class=\\"form-input\\"><label><strong>Password:</strong> <input value=\\"\\"></label></div><div><button>Log in</button></div></form></div>"`;
